bindings/java/src/main/com/apple/foundationdb/tuple/TupleUtil.java

```diff
@@ -567,35 +623,43 @@ static EncodeResult encodeAll(List<Object> items, byte[] prefix, List<byte[]> en
 			lenSoFar += result.totalLength;
 		}
 		//System.out.println("Joining whole tuple...");
-		return new EncodeResult(lenSoFar, versionPos);
+		return ENCODE_RESULT_THREAD_LOCAL.get().set(lenSoFar, versionPos);
 	}
 
 	static byte[] pack(List<Object> items, byte[] prefix) {
-		List<byte[]> encoded = new ArrayList<>(2 * items.size() + (prefix == null ? 0 : 1));
-		EncodeResult result = encodeAll(items, prefix, encoded);
+		ByteArrayOutputStream baos = new ByteArrayOutputStream(16 * items.size());
+		EncodeResult result = null;
+		try {
+			result = encodeAll(items, prefix, baos);
+		} catch (IOException e) {
+			throw new AssertionError();
```

Comment:
This should not be an `AssertionError`. I think a `RuntimeException` might be more appropriate unless there is something more specific than that that you can think of. It also should thread through the `IOException` it got as a cause.

Reply:
The thinking it is that both of these ought to be AssertionErrors because there is no path that leads to an IOException since the underlying stream is a ByteArrayOutputStream.